helper/ssh/provisioner.go

```diff
@@ -46,6 +49,12 @@ type SSHConfig struct {
 	TimeoutVal time.Duration `mapstructure:"-"`
 }
 
+func (c *SSHConfig) RemotePath() string {
+	return strings.Replace(
+		c.ScriptPath, "%RAND%",
+		strconv.FormatInt(int64(rand.Int31()), 10), -1)
+}
```

Comment:
Probably a bit of potential for future confusion that this isn't memoized, but this area of the code is simple enough that I think it's not a big deal.